### DISCUSSION TITLE
exec() Seems to Spawn PowerShell Window

### DISCUSSION THREAD

**Discussion by arrmo** [pyinstaller/pyinstaller#7591] [PyInstaller] [answered]
Hi,

Sorry if this is in the wrong discussion group - wasn't sure if I should enter it here or Help. But that aside ... having issues with my .exe on Windows - where I use the code below to call (execute) a Python command, via a passed string,
```
            with contextlib.redirect_stdout(cmdout):
                exec(self.label['value'])
            txtupdate = cmdout.getvalue()
```

I have console disabled in PyInstaller, and this code works fine from PyCharm, but ... with the .exe, I see a Terminal being opened when this code runs. Is this a known issue, or perhaps there is a way around this?

Thanks!

**Comment by bwoodsend** [comment, marked as answer]
What `exec()` is this? The builtin [exec()](https://docs.python.org/3/library/functions.html#exec) or some custom subprocess thing? The builtin `exec()` won't be responsible for creating new consoles unless the string you're passing to it contains subprocess code with the *create new console* flags set.

**Comment by arrmo** [comment]
I think you're on to something! Yes, I'm using the built in exec(), like you note. One of the sample commands I'm running is,
```
    psnetif = 'Get-NetIPInterface -AddressFamily IPv4 -Dhcp Enabled -ConnectionState Connected ' \
            '| Sort-Object -Property InterfaceMetric | Format-Table -HideTableHeaders'
    psifrun = subprocess.run("powershell -Command " + psnetif, capture_output=True, shell=False)
```

It does run PowerShell, but no shell => in pyCharm, no window is opened, but in the .exe ... I get a window. Thoughts? Or perhaps the create new console flags you mention? I need to dig on that one 😆.

Thanks!

**Comment by arrmo** [comment]
Found it - thanks to your lead, appreciate it! 🍻 

I added / changed,
```
psifrun = subprocess.run("powershell -Command " + psnetif, capture_output=True, creationflags=subprocess.CREATE_NO_WINDOW)
```

Sharing, hope to help others.

Thanks again.
